### PR TITLE
Raise WordPress required version to 6.4

### PR DESCRIPTION
**Pull request by renatonascalves**
Trac ticket: https://buddypress.trac.wordpress.org/ticket/9225

---
**This Pull Request is for code review only. Please keep all other discussion in the BuddyPress Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the WordPress Core Handbook for more details.**

### CODE REVIEW

File: bp-loader.php [hunk 1]
```diff
@@ -19,20 +19,20 @@
  * License URI:       https://www.gnu.org/licenses/gpl-2.0.html
  * Text Domain:       buddypress
  * Domain Path:       /bp-languages/
- * Requires PHP:      5.6
- * Requires at least: 6.1
+ * Requires PHP:      7.0
```
Comment (renatonascalves) [inline bp-loader.php:22]: According to [WordPress stats](https://wordpress.org/about/stats/), PHP 7.0 has less than 2% of usage. PHP 7.3 has around ~3.8%.

And PHP 7.4, 37.6%. **I personally would like to drop support for <7.3**. Meaning that BuddyPress would require 7.4>.

It's not clear to me why we were supporting 5.6, but we don't have data specific to BuddyPress, as far as I know.

Comment (renatonascalves) [reply bp-loader.php:22]: 7.0 is the minumum version supported by 6.4, that's why I defaulted to 7.0.

File: .github/workflows/unit-test-object.yml [hunk 2]
```diff
@@ -11,14 +11,34 @@ on:
...
-        object: ["false", "memcached", "redis"]
+        include:
+          - php: '8.3'
+            wp_version: 'trunk'
+            object: 'redis'
+            multisite: true
+          - php: '8.3'
+            wp_version: 'trunk'
+            object: 'memcached'
+            multisite: false
```
Comment (renatonascalves) [inline .github/workflows/unit-test-object.yml:25]: Those two are essentially to test with the object cache services.

File: .github/workflows/unit-test-object.yml [hunk 3]
```diff
@@ -11,14 +11,34 @@ on:
...
+            object: ''
+            multisite: false
+          - php: '8.2'
+            wp_version: 'latest'
+            object: ''
+            multisite: false
+          - php: '8.1'
+            wp_version: 'latest'
+            object: ''
+            multisite: false
```
Comment (renatonascalves) [inline .github/workflows/unit-test-object.yml:37]: PHP 8.1 and 8.2 are used the most by WordPress installations: https://wordpress.org/about/stats/

**Comment by renatonascalves** [comment]
@imath Could you take a look at the matrix here? I tried to balance our need with the Github Action limits. We want to run as many scenarios as possible, but at the same time, we don't have unlimited resources.

File: .github/workflows/unit-test-object.yml [hunk 4]
```diff
@@ -11,14 +11,34 @@ on:
...
+          - php: '7.4'
+            wp_version: 'latest'
+            object: ''
+            multisite: false
```
Comment (renatonascalves) [inline .github/workflows/unit-test-object.yml:41]: 7.4 accounts for 37% of all WordPress installations. Maybe we should test this more?!

File: bp-loader.php (hunk 1 above)
Comment (imath) [reply bp-loader.php:22]: I believe we should follow WordPress requirements about this. Raising PHP version according to the minimum required WP version: WP 6.4 => PHP 7.0 is fine with me.
WP 6.6 is supporting back to [7.2](https://core.trac.wordpress.org/browser/branches/6.6/src/wp-includes/version.php)

File: src/readme.txt [hunk 5]
```diff
@@ -4,10 +4,10 @@ Donate link:       https://wordpressfoundation.org
 Tags:              community, group, members, network, self-hosted
 License:           GNU General Public License v2 or later
 License URI:       https://www.gnu.org/licenses/gpl-2.0.html
-Requires PHP:      5.6
-Requires at least: 6.1
+Requires PHP:      7.0
+Requires at least: 6.4
 Tested up to:      6.6
-Stable tag:        14.0.0
+Stable tag:        15.0.0-alpha
```
Comment (imath) [inline src/readme.txt:10]:
```suggestion
Stable tag:        14.0.0
```
Please do not change stable tag. This only happens when building a new minor/major version. See https://github.com/buddypress/buddypress/blob/master/docs/contributor/project/release/build-checklist.md#version-bumps-in-dev

File: src/readme.txt [hunk 6]
```diff
@@ -54,15 +54,15 @@ Growing the BuddyPress community means better software for everyone!
 
 To run BuddyPress, we recommend your host supports:
 
-* PHP version 7.4 or greater.
+* PHP version 8.1 or greater.
```
Comment (imath) [inline src/readme.txt:57]:
```suggestion
* PHP version 7.4 or greater.
```

Let's follow how WordPress evolves here too: [7.4 is still minimum recommended PHP version](https://wordpress.org/about/requirements/). Let's keep it this way.

File: src/readme.txt [hunk 7]
```diff
@@ -54,15 +54,15 @@ Growing the BuddyPress community means better software for everyone!
...
 * MySQL version 5.7 or greater OR MariaDB version 10.4 or greater.
```
Comment (imath) [inline src/readme.txt:58]:
```suggestion
* MySQL version 8.0 or greater OR MariaDB version 10.5 or greater.
```

I think we should also follow WordPress db recommended versions.

File: .github/workflows/unit-test-object.yml (hunk 4 above)
Comment (imath) [inline .github/workflows/unit-test-object.yml:41]:
```suggestion
          - php: '7.4'
            wp_version: '6.4'
            object: ''
            multisite: false
         - php: '7.4'
            wp_version: '6.4'
            object: ''
            multisite: true
```
I think we should test PHP 7.4 with WP 6.4 + add a job to test multisite.

**Review by imath — CHANGES_REQUESTED** [review]
Thanks a lot for your work on it @renatonascalves I've added some comments in the code. I think we should follow WordPress recommended versions and choose PHP required version according to the first WP version we support.

File: src/readme.txt (hunk 5 above)
Comment (renatonascalves) [reply src/readme.txt:10]: Ops! Good catch!

File: src/readme.txt (hunk 6 above)
Comment (renatonascalves) [reply src/readme.txt:57]: This is actually the recommended PHP version, not the minumum.

Comment (renatonascalves) [reply src/readme.txt]: Ops. I see. That's correct!

File: .github/workflows/unit-test-object.yml [hunk 8]
```diff
@@ -11,14 +11,38 @@ on:
...
+          - php: '7.4'
+            wp_version: '6.4'
+            object: ''
+            multisite: false
+         - php: '7.4'
+            wp_version: '6.4'
+            object: ''
+            multisite: true
```
Comment (renatonascalves) [inline .github/workflows/unit-test-object.yml:45]:
```suggestion
          - php: '7.4'
            wp_version: '6.4'
            object: ''
            multisite: false
          - php: '7.4'
            wp_version: '6.4'
            object: ''
            multisite: true
```

**Review by imath — APPROVED** [review]
Thanks a lot! All lights green to me, feel free to SVN commit it 💪

**Comment by renatonascalves** [comment]
Thanks! I'll publish a post in the bpdevel blog too about it. 🎈

**Pull request closed without merging**